plugins/modules/foreman_templates_import.py

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) 2020 Anton Nesterov (@nesanton)
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: foreman_templates_import
+short_description: Sync templates from a repository
+description:
+  - Sync provisioning templates, report_templates, partition tables and job templates from external git repository and/or file system.
+  - Based on foreman_templates plugin U(https://github.com/theforeman/foreman_templates).
+  - Some defaults can be set in TemplateSync settings using M(foreman_setting) or GUI.
+  - Module attempts to be idempotent as much as the plugin allows.
+author:
+  - "Anton Nesterov (@nesanton)"
+options:
+  prefix:
+    description:
+      - Adds specified string to beginning of the template, but only if the template name does not start with the prefix already.
```

Comment:
maybe sthing like:
...of all imported templates that do not yet have that prefix...

If i remember correctly, this could be set in the settings. If that is so, maybe drop a note here, or collect the names of all such options in the note above.

Reply:
I tried to keep the description consistent with what they say in plugin docs here https://www.theforeman.org/plugins/foreman_templates/6.0/index.html

For many params I changed it, and I agree that it's too long, so ok!

Reply:
About settings, I'm saying this in line 33
`- Some defaults can be set in TemplateSync settings using M(foreman_setting) or GUI.`

Reply:
I was wondering if it's worth beeing more specific about which settings are.
It's probably not "All settings can..."

Reply:
Actually they are all there indeed. Changing to:
`- Default values for all module options can be set using M(foreman_setting) for TemplateSync category or on the settings page in GUI.`

Reply:
@evgeni, do you want this in `notes:`?

Reply:
If it applies to all parameters, then yes, notes seems appropriate.

Oh, and could you please write UI or WebUI, not GUI? ;)

Reply:
there you go